stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -2476,12 +2480,39 @@ internal constructor(
         channelsIds: List<String>,
         lastSyncAt: Date,
     ): Call<List<ChatEvent>> {
-        return api.getSyncHistory(channelsIds, lastSyncAt)
+        val stringDate = streamDateFormatter.format(lastSyncAt)
+
+        return api.getSyncHistory(channelsIds, stringDate)
             .withPrecondition(scope) {
                 checkSyncHistoryPreconditions(channelsIds, lastSyncAt)
             }
     }
 
+    /**
+     * Returns all events that happened for a list of channels since last sync (while the user was not
+     * connected to the web-socket). LastSyncAt is in the format yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'. Use this version of
+     * getSyncHistory when high precision is necessary.
+     *
+     * @param channelsIds The list of channel CIDs. Cannot be empty.
+     * @param lastSyncAt The last time the user was online and in sync. Shouldn't be later than 30 days.
+     *
+     * @return Executable async [Call] responsible for obtaining missing events.
+     */
+    @CheckResult
+    public fun getSyncHistory(
+        channelsIds: List<String>,
+        lastSyncAt: String,
+    ): Call<List<ChatEvent>> {
+        val parsedDate = streamDateFormatter.parse(lastSyncAt) ?: throw IllegalArgumentException(
+            "The string for data: $lastSyncAt could not be parsed for format: ${streamDateFormatter.datePattern} "
+        )
```

Comment:
maybe we could make `getSyncHistory` with a `String` param as an `internal` function to avoid having this validation by parsing the input date? This is because we can internally guarantee we always gonna pass a valid `rawDate`

Reply:
It is not possible to make it internal because this is uses in state module, but I put the annotation of `InternalStreamChatApi`